airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -365,7 +365,10 @@ def execute(self, context) -> Optional[str]:
                 self.log.info("creating pod with labels %s and launcher %s", labels, launcher)
                 final_state, _, result = self.create_new_pod_for_operator(labels, launcher)
             if final_state != State.SUCCESS:
-                status = self.client.read_namespaced_pod(self.pod.metadata.name, self.namespace)
+                if self.is_delete_operator_pod:
+                    status = final_state
```

Comment:
I wonder if we should make this actually dump the whole pod like it would if `is_delete_operator_pod` was false, instead of the state. It would take a little refactoring to make it happen, but seems like it could be pretty helpful in that scenario?

Maybe this exception should always just include the state, then have `create_new_pod_for_operator` log the full pod on failure?

Reply:
That's a good idea - we could modify `monitor_pod` to return a Tuple of `[State, event, result]` instead of just `[State, result]` (https://github.com/apache/airflow/blob/b844621a6f42cc20b3103bc70c774023268f8de8/airflow/providers/cncf/kubernetes/utils/pod_launcher.py#L170).  Then `create_new_pod_for_operator` could pass that back through to `execute` too (or `create_new_pod_for_operator` could read the pod again in it's `finally` clause and pass that back)

Reply:
Okay, I've updated the code to return the full V1Pod back through from `monitor_pod`.  This allows Airflow to log the full pod description in the event of a failure regardless of `is_delete_pod_operator`. 

I modified the original test that checked whether `read_namespaced_pod` was called in the event of a failure, since it doesn't need to be called either way now - it just checks that the failed pod description is reported in the Exception.